src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -290,56 +289,56 @@ private void RemoveDirectoryInternal(string fullPath, bool recursive, bool throw
 
             while (Interop.libc.rmdir(fullPath) < 0)
             {
-                int errno = Marshal.GetLastWin32Error();
-                switch (errno)
+                Interop.ErrorInfo errorInfo = Interop.System.GetLastErrorInfo();
+                switch (errorInfo.Error)
                 {
-                    case Interop.Errors.EINTR: // interrupted; try again
+                    case Interop.Error.EINTR: // interrupted; try again
                         continue;
-                    case Interop.Errors.EACCES:
-                    case Interop.Errors.EPERM:
-                    case Interop.Errors.EROFS:
-                    case Interop.Errors.EISDIR:
+                    case Interop.Error.EACCES:
+                    case Interop.Error.EPERM:
+                    case Interop.Error.EROFS:
+                    case Interop.Error.EISDIR:
                         throw new IOException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, fullPath)); // match Win32 exception
-                    case Interop.Errors.ENOENT:
+                    case Interop.Error.ENOENT:
                         if (!throwOnTopLevelDirectoryNotFound)
                         {
                             return;
                         }
                         goto default;
                     default:
-                        throw Interop.GetExceptionForIoErrno(errno, fullPath, isDirectory: true);
+                        throw Interop.GetExceptionForIoErrno(errorInfo, fullPath, isDirectory: true);
                 }
             }
         }
 
         public override bool DirectoryExists(string fullPath)
         {
-            int errno;
-            return DirectoryExists(fullPath, out errno);
+            Interop.ErrorInfo errorInfo;
```

Comment:
Maybe ```ignored``` instead of ```errorInfo```?  I'd previously named it ```erno``` because ```int``` isn't self-descriptive as to what's being ignored, but now that it's ```Interop.ErrorInfo```, it's much clearer.